File: VirtualJudgecontest3DataStructures120242/Bsaltando.py
while True:
    try:
        a = [int(x) for x in input().split()]
    except EOFError:
        break    

    numbers = [False for _ in range(a[0] + 1)]
    
    valid = True
    
    for i in range(1, len(a) - 1):
        diff = abs(a[i] - a[i + 1])
        if diff > a[0] - 1 or (numbers[diff]):
            valid = False
        else:
            numbers[diff]
    
    if valid:
        print('Jolly')
    else:
        print('Not jolly')

File: VirtualJudgecontest3DataStructures120242/Bsaltando.cpp
// time-limit: 1000
// problem-url: https://vjudge.net/contest/653845#problem/B
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef long l;
 
template<typename T>
void print(vector<T> &vec){
    for(auto &x:vec) cout<<x<<' ';
    cout<<'\n';
}
template<typename T>
void print(vector<pair<T,T>> &vec){
    for(auto &x:vec) cout<<x.first<<' '<<x.second<<'\n';
}
void barra(){ cout<<"----------------------"<<'\n'; }
 
 
void solve(){
    int n; cin >> n;
    vector<bool> numbers(n);
    for (int i = 0; i < n + 1; i++) {
        numbers[i] = false;
    }
    bool valid = true;
    int n1, n2;
    cin >> n1;
    for (int i = 1; i < n; i++) {
        cin >> n2;
        if (numbers[abs(n1 - n2)]) {
            valid = false;
        } else {
            numbers[abs(n1 - n2)] = true;
        }
        n1 = n2;
    }
    if (valid) {
        cout << "Jolly" << endl;
    } else {
        cout << "Not jolly" << endl;
    }
}
 
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    int t=1;
    //cin>>t;
    while(t--) solve(); 
}